tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSDesugaring.scala

```diff
@@ -2279,14 +2279,74 @@ private[emitter] class JSDesugaring(internalOptions: InternalOptions) {
 
   private[emitter] def genLoadJSFromSpec(spec: JSNativeLoadSpec)(
       implicit outputMode: OutputMode, pos: Position): js.Tree = {
+
+    def pathSelection(from: js.Tree, path: List[String]): js.Tree = {
+      path.foldLeft(from) {
+        (prev, part) => genBracketSelect(prev, js.StringLiteral(part))
+      }
+    }
+
     spec match {
       case JSNativeLoadSpec.Global(path) =>
-        path.foldLeft(envField("g")) {
-          (prev, part) => genBracketSelect(prev, js.StringLiteral(part))
+        pathSelection(envField("g"), path)
+
+      case JSNativeLoadSpec.Import(module, path) =>
+        val moduleValue = envModuleField(module)
+        path match {
+          case DefaultExportName :: rest =>
+            val defaultField = genCallHelper("moduleDefault", moduleValue)
```

Comment:
This is a bit bothersome, but frankly I also don't know how to do this better...